### ISSUE TITLE
Can concourse deploy to aws?

### ISSUE THREAD

**Issue by erichelin** [concourse/concourse#1236]
Not sure if this the best location, but I was curious if concourse can deploy to aws (probably elastic beanstalk)?  I did some research online and didn't find any definitive answer either way.  Currently, we use our concourse with cloud foundry.

**Comment by vito** [comment]
Do you mean can concourse be deployed to AWS? If so, yes: http://concourse.ci/installing.html

Do you mean can concourse deploy something to AWS? If so, this would probably be done by a resource. Maybe https://github.com/pivotal-cf-experimental/cloudformation-resource ?

Concourse itself doesn't integrate with anything, it just supports resources, which provide a way to integrate with anything. If there's not one for Elastic Beanstalk specifically, it could be implemented.

**Comment by engrun** [comment]
I'm also interested in this topic. Can concourse be installed on AWS **without** BOSH? We're not using BOSH at the moment (and not planning to), and I'm rather new to AWS, so it seems a bit of a too steep learning curve just to install a CI/CD server. 
I like the concepts of Concourse and would like to try it out, however, I can't find any official docs on how to install to cloud providers without using BOSH. 
I found a description here - https://github.com/mumoshu/concourse-aws - however, depending on a 3rd party "hack" is not very tempting. 

My guess would be that requiring BOSH just for Concourse could be a blocker for some people (or I might be wrong, as mentioned, I'm new to this)

**Comment by pivotal-topher-bullock** [comment]
@engrun 
**TL;DR:** BOSH is not explicitly required to run Concourse on AWS ( or any IaaS ). If you just want to get the Concourse Web UI and a Worker up on AWS, I'd suggest running the [binaries](http://concourse.ci/binaries.html) on EC2 instances (and optionally work your way up to EBS); one instance running `concourse web` and another (or several others) running `concourse worker`.

---

You can use whatever infrastructure automation tools / process you're comfortable with to get the binaries up to the EC2 instance and keep them running. That will reduce the cognitive overhead to learn AWS, Concourse, and BOSH all at once which would be a daunting task. 

We suggest and document using BOSH to  for two reasons: 
 1. BOSH is great at keeping processes and VMs running and healthy (it uses monit for processes and has a health monitor for vms)
 1. Internally on the team we have a lot of knowledge and support around BOSH, since it's a CF project and staffed by Pivotal

For the first point, you're free to use other methods to keep the binaries up (the BOSH release aspect is just a lot of fancy-dancy magic around keeping a process up and the VM its running in up ) .. Chef, Puppet, whatevs. For the second point, BOSH has an [infamously steep learning curve](http://imgur.com/4UpbgJm) compared to other tools, but personally, having sweat through it, I think it's a worthwhile effort for the cloud lifecycle management niceties that BOSH provides.

**Comment by engrun** [comment]
Ok, so we have installed Concourse on AWS.  One `web` and two `worker` .  However, the next step is to configure autoscaling. This sort of works, in the sense that both `worker` and `cluster instances` are scaled out and then in (based on CPU). Scaling out is working fine. The problem is that AWS does not know which `worker` tasks to scale in. The consequence is that the AWS can possibly kill a `worker` task that is currently busy. The same goes for the EC2 instances. AWS can possibly terminate an instance that is busy. ( we believe it works like this, but we're AWS novices)

**Comment by niklongstone** [comment]
Hi @engrun you can use the AWS Instance Protection to decide when the instance is ready or not for termination: http://docs.aws.amazon.com/autoscaling/latest/userguide/as-instance-termination.html#instance-protection
Probably you can implement a function that checks registered workers and enable or disable the instance protection accordingly.
